src/qibocal/protocols/two_qubit_interaction/snz_optimize.py

```diff
@@ -0,0 +1,364 @@
+    angles: dict
+    """Native SNZ angle."""
+
+    def __contains__(self, key: QubitPairId):
+        """Check if key is in class.
+
+        Additional  manipulations required because of the Results class.
+        """
+        pairs = {
+            (target, control) for target, control, _, _, _ in self.fitted_parameters
+        }
+        return key in pairs
+
+
+OptimizeTwoQubitGateType = np.dtype(
+    [
+        ("amp", np.float64),
+        ("theta", np.float64),
+        ("rel_amplitude", np.float64),
+        ("prob_target", np.float64),
+        ("prob_control", np.float64),
+    ]
+)
+
+
+@dataclass
+class SNZFinetuningData(Data):
+    data: dict[tuple, npt.NDArray[OptimizeTwoQubitGateType]] = field(
+        default_factory=dict
+    )
+    """Raw data."""
+    amplitudes: dict[tuple[QubitId, QubitId], float] = field(default_factory=dict)
+    """"Amplitudes swept."""
+    rel_amplitudes: list[float] = field(default_factory=list)
+    """Durations swept."""
+    angles: list = field(default_factory=list)
+
+    def __getitem__(self, pair):
+        """Extract data for pair."""
+        return {
+            index: value
+            for index, value in self.data.items()
+            if set(pair).issubset(index)
+        }
+
+    @property
+    def swept_virtual_phases(self):
+        """List of swept phases."""
+        return [-1 * i for i in self.angles]
+
+    def register_qubit(
+        self, target, control, setup, ratio, theta, amp, prob_control, prob_target
+    ):
+        """Store output for single pair."""
+        size = len(theta) * len(amp)
+        amplitude, angle = np.meshgrid(amp, theta, indexing="ij")
+        ar = np.empty(size, dtype=OptimizeTwoQubitGateType)
+        ar["theta"] = angle.ravel()
+        ar["amp"] = amplitude.ravel()
+        ar["prob_control"] = prob_control.ravel()
+        ar["prob_target"] = prob_target.ravel()
+        self.data[target, control, setup, ratio] = np.rec.array(ar)
+
+
+def _aquisition(
+    params: SNZFinetuningParamteters,
+    platform: CalibrationPlatform,
+    targets: list[QubitPairId],
+) -> SNZFinetuningData:
+    """Acquisition for the optimization of SNZ amplitudes."""
+    ratio_range = np.arange(
+        params.amp_ratio_min, params.amp_ratio_max, params.amp_ratio_step
+    )
+    data = SNZFinetuningData()
+    data.rel_amplitudes = ratio_range.tolist()
+    data.angles = np.arange(
+        params.theta_start, params.theta_end, params.theta_step
+    ).tolist()
+    for pair in targets:
+        ordered_pair = order_pair(pair, platform)
+        flux_channel = platform.qubits[ordered_pair[1]].flux
+        target_vz = pair[0]
+        other_qubit_vz = pair[1]
+        # Find CZ flux pulse
+        cz_sequence = getattr(platform.natives.two_qubit[ordered_pair], "CZ")()
+        flux_channel = platform.qubits[ordered_pair[1]].flux
+
+        for cz_pulse in cz_sequence:
+            if cz_pulse[0] == flux_channel:
+                flux_pulse = cz_pulse[1]
```

Comment:
We could skip the for loop by using the [`channel`](https://github.com/qiboteam/qibolab/blob/20ef5432963a975d5aba5bd699eb811153ac9416/src/qibolab/_core/sequence.py#L79-L81) method in `PulseSequence`
```suggestion
        cz_sequence = platform.natives.two_qubit[ordered_pair].CZ
        flux_channel = platform.qubits[ordered_pair[1]].flux
        flux_pulses = cz_sequence.channel(flux_channel)
        assert len(flux_pulses) == 1, "Only 1 flux pulse is supported"
        flux_pulse = flux_pulses[0]
```